File: wms-web/src/constants/stockStatus.ts
export const STOCK_STATUS_OPTIONS = [
  { value: 'AVAILABLE', label: '可用' },
  { value: 'FROZEN', label: '冻结' },
  { value: 'INSPECTION', label: '待检' },
  { value: 'UNQUALIFIED', label: '不合格' },
  { value: 'SCRAPPED', label: '报废' },
] as const

export const STOCK_STATUS_LABEL: Record<string, string> = Object.fromEntries(
  STOCK_STATUS_OPTIONS.map((item) => [item.value, item.label]),
)

export function stockStatusLabel(code?: string | null) {
  if (!code) return '-'
  return STOCK_STATUS_LABEL[code] || code
}

File: wms-backend/src/main/java/com/wms/inventory/constant/StockStatus.java
package com.wms.inventory.constant;

import lombok.Getter;

import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;

/**
 * 库存状态（与 inventory.stock_status 字段一致）。
 */
@Getter
public enum StockStatus {

    AVAILABLE("AVAILABLE", "可用"),
    FROZEN("FROZEN", "冻结"),
    INSPECTION("INSPECTION", "待检"),
    UNQUALIFIED("UNQUALIFIED", "不合格"),
    SCRAPPED("SCRAPPED", "报废");

    private final String code;
    private final String label;

    StockStatus(String code, String label) {
        this.code = code;
        this.label = label;
    }

    public static Optional<StockStatus> fromCode(String code) {
        if (code == null || code.isBlank()) {
            return Optional.empty();
        }
        String normalized = code.trim().toUpperCase();
        return Arrays.stream(values())
                .filter(s -> s.code.equals(normalized))
                .findFirst();
    }

    public static String labelOf(String code) {
        return fromCode(code).map(StockStatus::getLabel).orElse(code);
    }

    public static List<Map<String, String>> options() {
        return Arrays.stream(values())
                .map(s -> {
                    Map<String, String> item = new LinkedHashMap<>();
                    item.put("value", s.code);
                    item.put("label", s.label);
                    return item;
                })
                .toList();
    }
}
